### PR TITLE
sci-libs/opencascade: Fix build on musl

### PR DESCRIPTION
**Pull request by vimproved**
Fixes building and unmasks opencascade on musl. Conditionalized use of GNU extensions behind the __GLIBC__ macro (and execinfo.h behind a cmake configure step for those using libexecinfo from an overlay). Patch adapted from alpine.

### CODE REVIEW

File: sci-libs/opencascade/files/opencascade-7.7.0-musl.patch [hunk 1]
```diff
@@ -0,0 +1,250 @@
+diff -upr a/CMakeLists.txt b/CMakeLists.txt
```
Comment (thesamesam) [inline sci-libs/opencascade/files/opencascade-7.7.0-musl.patch:1]: This patch is pretty big and should really go upstream first.

Comment (vimproved) [reply sci-libs/opencascade/files/opencascade-7.7.0-musl.patch:1]: Understood, I'll submit a PR upstream.

Comment (thesamesam) [reply sci-libs/opencascade/files/opencascade-7.7.0-musl.patch:1]: Thanks!

Comment (vimproved) [reply sci-libs/opencascade/files/opencascade-7.7.0-musl.patch:1]: @thesamesam Upstream requires me to sign a CLA with my Mailing Address, which I'm not particularly comfortable giving out. Any alternatives?

**Comment by vimproved** [comment]
Made the patch considerably smaller, could this be looked at again?

**Review by juippis — APPROVED** [review]
It at least builds, but I get the following in musl:
```
 * QA Notice: Package triggers severe warnings which indicate that it
 *            may exhibit random runtime failures.
 * /var/tmp/portage/sci-libs/opencascade-7.7.0-r2/work/occt-185d29b/src/StepToTopoDS/StepToTopoDS_TranslateShell.cxx:143:27: warning: 'this' pointer is null [-Wnonnull]
```

**Pull request closed without merging**